ophyd/areadetector/plugins.py

```diff
@@ -836,6 +837,28 @@ class JPEGPlugin(FilePlugin, version=(1, 9, 1), version_type='ADCore'):
     jpeg_quality = Cpt(SignalWithRBV, 'JPEGQuality', kind='config')
 
 
+@register_plugin
+class KafkaPlugin(PluginBase, version=(1, 9, 1), version_type='ADCore'):
+
+    _default_suffix = 'KAFKA1:'
+    _suffix_re = r'KAFKA\d:'
+    _html_docs = ['NDPluginKafka.html']
+    _plugin_type = 'NDPluginKafka'
+
+    source_name = Cpt(SignalWithRBV, 'SourceName')
+    reconnect_flush = Cpt(SignalWithRBV, 'ReconnectFlush')
+    reconnect_flush_time = Cpt(SignalWithRBV, 'ReconnectFlushTime')
+    kafka_buffer_size = Cpt(SignalWithRBV, 'KafkaBufferSize')
+    kafka_max_message_size = Cpt(SignalWithRBV, 'KafkaMaxMessageSize')
+    unsent_packets = Cpt(EpicsSignalRO, 'UnsentPackets_RBV')
+    connection_status = Cpt(EpicsSignalRO, 'ConnectionStatus_RBV')
+    connection_message = Cpt(EpicsSignalRO, 'ConnectionMessage_RBV')
+    kafka_topic = Cpt(SignalWithRBV, 'KafkaTopic')
+    kafka_broker_address = Cpt(SignalWithRBV, 'KafkaBrokerAddress')
+    kafka_stats_interval_time = Cpt(SignalWithRBV, 'KafkaStatsIntervalTime')
+    kafka_max_queue_size = Cpt(SignalWithRBV, 'KafkaMaxQueueSize')
```

Comment:
I'd personally sort the entries alphabetically for an easier search of a parameter, but it's up to you. If it's based on the order from the AD source, we can leave it as is.

Reply:
We can make them alphabetical if that's preferable - as you said, the current order is just the order they were defined in the Plugin Driver code.

Reply:
I am in favor of the sorted items, or, at least, the same order in which the parameters appear in CSS screens.